fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+- **Index - Add or Remove or Update**
+    - As indices are removed from the table definition, the removed indices must be manually dropped for each Resource table. 
+    - As indices are added to the table definition, the version of the table must be updated, and the index must be applied and updated manually.
+- **Constraint Updates** If there are Foreign Key updates, the changes must be applied manually and reflected in the code base. 
+- **Search Parameter Changes (Specification and Tenant)**
+    - If the the parameter type or code is changed, the PARAMETERS_NAME and the corresponding table must be updated to remove references to the removed parameter (based on code). 
+    - If there is a new SearchParameter with `code` newly added to the system, the Search Parameter value entry is only updated when the resource is updated. 
+
+For each of the above changes, the version of the table must be incremented. 
+
+The resource table stores the FHIR resource as a compressed blob, changes to the specification or extensions should only impact Search values. 
+
+The tables have various indices - PrimaryKey, Index and UniqueIndex. These indices are created as part of the Java object - Table.  These 
+
+If a modification is made to a Search Parameter (addition, removal), then the resource uses the existing search parameter values until a new version of the resource is processed.
+
+Each of these tables has row-level permissions based on the conditional READ-only global variable, such as: 
+
+``` sql
+CREATE PERMISSION FHIRAPP.AUDITEVENT_COMPOSITES_TENANT
+    ON FHIRAPP.AUDITEVENT_COMPOSITES FOR ROWS
+    WHERE FHIRAPP.AUDITEVENT_COMPOSITES.MT_ID = FHIR_ADMIN.SV_TENANT_ID 
+    ENFORCED FOR ALL ACCESS ENABLE ;
+```
+
+These permissions should not be removed, migrated, altered or changed.
+
+## Managing Stored Procedures
+
+`FHIRDATA.ADD_CODE_SYSTEM` is created in the default schema.  The stored procedures `add_code_system`, `add_parameter_name`, `add_resource_type`, and `add_any_resource`.
+
+If updates are made to the `add_code_system.sql`, the `FhirSchemaGenerator.buildSchema`'s FhirSchemaConstants.INITIAL_VERSION must be updated to the next highest integer (in this case 2). 
+
+``` java
+ProcedureDef pd = model.addProcedure(this.schemaName, 
+    ADD_CODE_SYSTEM, 
+    FhirSchemaConstants.INITIAL_VERSION, 
+    () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_CODE_SYSTEM.toLowerCase() + ".sql", null), 
+    Arrays.asList(fhirSequence, codeSystemsTable, allTablesComplete), 
+    procedurePrivileges);
+```
+
+When the `fhir-persistence-schema` buildSchema is executed, the **INSERT**, **SELECT**, **UPDATE**, **DELETE** grant is applied again, and the procedures are updated. Each Procedure privilege is reset upon re-executing the action. 
```

Comment:
What is "buildSchema" ?   I don't see that one mentioned at https://github.com/IBM/FHIR/tree/master/fhir-persistence-schema#execute-the-fhir-persistence-schema-command-line-interface-cli ...is it the same as the `update-proc` action?

Reply:
I clarified